module/platform/Webots/src/Webots.cpp

```diff
@@ -327,107 +327,122 @@ namespace module::platform {
+
+                                if (n >= 0) {
+                                    if (initial_message.data() == std::string("Welcome")) {
+                                        // good
+                                        log<NUClear::INFO>(
+                                            fmt::format("Connected to {}:{}", server_address, server_port));
+                                    }
+                                    else if (initial_message.data() == std::string("Refused")) {
+                                        log<NUClear::FATAL>(
+                                            fmt::format("Connection to {}:{} refused: your IP is not white listed.",
+                                                        server_address,
+                                                        server_port));
+                                        // Halt and don't retry as reconnection is pointless.
+                                        close(fd);
+                                        powerplant.shutdown();
+                                    }
+                                    else {
+                                        log<NUClear::FATAL>(fmt::format("{}:{} sent unknown initial message",
+                                                                        server_address,
+                                                                        server_port));
+                                        // Halt and don't retry as the other end is clearly not Webots
+                                        close(fd);
+                                        powerplant.shutdown();
+                                    }
+                                }
+                                else {
+                                    // There was nothing sent
+                                    log<NUClear::DEBUG>("Connection was closed.");
+                                    active_reconnect.store(false);
+                                    return;
+                                }
+
+                                // Set the real time of the connection initiation
+                                connect_time = NUClear::clock::now();
+                                // Reset the simulation connection time
+                                utility::clock::last_update = NUClear::base_clock::now();
+
+                                connection_active = true;
+
+                                // Clear socket && buffer
+                                unsigned long available = 0;
+                                if (::ioctl(fd, FIONREAD, &available) < 0) {
+                                    log<NUClear::ERROR>(
+                                        fmt::format("Error querying for available data, {}", strerror(errno)));
+                                    return;
+                                }
+                                const size_t old_size = buffer.size();
```

Comment:
Should we resize the buffer?

Reply:
If we don't how will we fit all the new data in? And the size is important later on. Also not relevant to this PR.

Reply:
Oh wait this is a different line. I don't see why, we're clearing it anyway?

Reply:
Tbh I'm not sure this is really needed anyway.

Reply:
What does this section do?

Reply:
It's clearing the socket and buffer when reconnecting but I don't think it's needed - testing now.

Reply:
I've got rid of it, it didn't seem to be needed.